### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by csoneson**
Unrelated to the k-mer changes; fixes to address issues raised by `BiocCheck`

### CODE REVIEW

File: R/binning.R [hunk 1]
```diff
@@ -108,3 +108,47 @@ bin <- function(x, binmode = c("equalN", "equalWidth", "breaks"),
...
+
+#' @export
+#' @rdname getSetZeroBin
+getZeroBin <- function(bins) {
+    attr(bins, "bin0")
+}
+
+#' @export 
+#' @rdname getSetZeroBin
+setZeroBin <- function(bins, zeroBin) {
```
Comment (mbstadler) [inline R/binning.R:140]: I have replaced `stopifnot()` by `.assertScalar` and `.assertVector` - we have them in `monaLisa` anyway and I think they give more informative error messages. I hope you are ok with that change.

File: R/motif_enrichment_monaLisa.R [hunk 2]
```diff
@@ -348,7 +354,28 @@
...
+                median(seq_lengths[df$isForeground & as.character(df$GCbin)==x])
+            }, 
+            FUN.VALUE = 0)
+        median_FG_length_per_GCbin_vector <- 
+            ifelse(df$isForeground,
+                   NA,
+                   median_FG_length_per_GCbin[as.character(df$GCbin)])
+        # ... correct background sequence weights
+        w_bg <- !df$isForeground
+        df$GCwgt[w_bg] <- df$GCwgt[w_bg] * seq_lengths[w_bg] / median_FG_length_per_GCbin_vector[w_bg]
```
Comment (mbstadler) [inline R/motif_enrichment_monaLisa.R:375]: Dania's tests show that the length correction does the right thing, my intuition is confused, though. For example, if a bg seq is twice as long as the median fg seq in that bin, it's weight will be doubled (not halfed). Somehow I was assuming that longer sequences should be downweighted relative to shorter ones??

**Review by mbstadler — COMMENTED** [review]
One general thing I am not sure about: There are many occurrences of something like: `attr(bins, "bin0")`.
Should we replace them by `getZeroBin(bins)`?
 It will probably create a bit of overhead that can be negleted, but using the interface might shield us from potential changes of the underlying implementation? On the other hand, the current code works and there are other attributes for which we do not provide any API at the moment.

File: R/binning.R (hunk 1 above)
Comment (csoneson) [reply R/binning.R:140]: Oh, yes - great!

**Comment by csoneson** [comment]
> One general thing I am not sure about: There are many occurrences of something like: `attr(bins, "bin0")`.
> Should we replace them by `getZeroBin(bins)`?
> It will probably create a bit of overhead that can be negleted, but using the interface might shield us from potential changes of the underlying implementation? On the other hand, the current code works and there are other attributes for which we do not provide any API at the moment.

Conceptually, my thought would be yes - replace by `getZeroBin(bins)`, for the reason you give above. However, there may be more pressing issues to prioritize.

**Review by mbstadler — APPROVED** [review]

**Pull request merged**